bin/birdhouse

```diff
@@ -188,6 +300,275 @@ parse_configs_args() {
   esac
 }
 
+# Source the birdhouse environment. This will only work the first time it is
+# called so that it can be called multiple times within the same process without
+# triggering a re-read of the configuration settings every time.
+source_env() {
+  if [ "$_BIRDHOUSE_ENV_ALREADY_SOURCED" != "true" ]; then
+    eval "$(print_config_command no-suffix)"
+    export _BIRDHOUSE_ENV_ALREADY_SOURCED=true
+  fi 
+}
+
+# Ensure that the birdhouse stack is running before executing another command.
+# If the stack is not running, this will raise a log ERROR message and exit.
+ensure_birdhouse_running() {
+  source_env
+  if ! docker ps --filter label=com.docker.compose.project --format '{{.Label "com.docker.compose.project"}}' | grep -q "${COMPOSE_PROJECT_NAME:-birdhouse}"; then
+      log ERROR "This command requires that the birdhouse stack be running. Start the stack first with '${THIS_BASENAME} compose up -d' and try again."
+      exit 1
+  fi
+  "$@"
+}
+
+# Backup or restore component data. This is called by the do_backup_restore function.
+# The BIRDHOUSE_BACKUP_RESTORE_COMMAND variable determines whether this will backup or restore.
+# The first argument determines whether to run a backup or restore job (must be one of 'backup' or 'restore')
+# The second argument determines which components will be backed up or restored. This argument is either a
+# space delimited string containing the names of components or the string '*' (selects all enabled components).
+do_component_backup_restore() {
```

Comment:
Why are these commands defined here, and not inside `lib/backup-restore.include.sh`?

I believe the various `parse_backup_args`, etc. make sense here since they are specific about CLI args parsing, but the various `do_...` that actually perform a command, or that are specific to `restic` should be in another file.

Reply:
The functions in `lib/backup-restore.include.sh` (now renamed to `lib/backup.include.sh`) are helper functions that are only used by the `backup` scripts (found in the component directories).

This function is a helper function used by `do_backup_create_restore` and was only created to DRY the code up a bit.

In other words: this function does not need to be called by the `backup` scripts so it shouldn't go in `lib/backup.include.sh`